### PR TITLE
Clean up Audio Player code and make Audio player look more like Figma spec

### PR DESCRIPTION
**Pull request by skywalkerdude**

### CODE REVIEW

File: Hymns/Audio Player/AudioPlayerView.swift [hunk 1]
```diff
@@ -2,11 +2,12 @@ import SwiftUI
 import AVFoundation
 import Combine
 
+// https://github.com/ChrisMash/AVPlayer-SwiftUI
+// https://medium.com/flawless-app-stories/avplayer-swiftui-part-2-player-controls-c28b721e7e27
+// TODO For some reason the combine stuff isn't working with our URLS that aren't straight up mp3 urls for example http://www.hymnal.net/en/hymn/h/894/f=mp3 that is coming from musicJson. However, the combine works when the url is a direct mp3 url such as https://www.hymnal.net/Hymns/NewSongs/mp3/ns0767.mp3
 struct AudioView: View {
 
-    //A second state variable play button toggle is necessary because currentlyplaying is also in control of graying out the music player and working with seeking. We don't want the music player to be greyed out and unavailable everytime we pause the music.
     @State var currentlyPlaying = false
```
Comment (skywalkerdude) [inline Hymns/Audio Player/AudioPlayerView.swift:10]: @finben This seems to be unnecessary and it looks like it's working. Please let me know this is incorrect, according to your understanding.

**Comment by skywalkerdude** [comment]
Yeah it does. I'm working on fixing that!

**Comment by finben** [comment]
Check my additions to add the looping functionality.

File: Hymns/Audio Player/AudioPlayerViewModel.swift [hunk 2]
```diff
@@ -3,11 +3,27 @@ import AVFoundation
...
+    func repeatingOn(audioPlayer: AVPlayer, looping: Bool) {
+        if looping {
+            observer = NotificationCenter.default.addObserver(forName: NSNotification.Name.AVPlayerItemDidPlayToEndTime, object: nil, queue: nil) { _ in
+                audioPlayer.seek(to: CMTime.zero)
+                audioPlayer.play()
+            }} else {
+            guard let observer = observer else {
+                return
+            }
+            NotificationCenter.default.removeObserver(observer)
```
Comment (skywalkerdude) [inline Hymns/Audio Player/AudioPlayerViewModel.swift:23]: @finben why do you need to remove the observer here?

**Comment by finben** [comment]
Need to remove the observer for if the user toggles repeat back off. If the
observer is not removed it will just keep popping

Luke <notifications@github.com> schrieb am Mo., 8. Juni 2020, 6:54 PM:

**Comment by skywalkerdude** [comment]
yup!

**Pull request merged**